diff --git a/dist/react/utils/useFileDiffInstance.js b/dist/react/utils/useFileDiffInstance.js
index 5775aaad39022671c29fad47f64f0a0c65bd4067..f8ea39e22064b2c80d5f9473ed8978fa771b9924 100644
--- a/dist/react/utils/useFileDiffInstance.js
+++ b/dist/react/utils/useFileDiffInstance.js
@@ -37,7 +37,8 @@ function useFileDiffInstance({ fileDiff, options, editorOptions, lineAnnotations
 				prerenderedHTML
 			});
 		} else {
-			if (instanceRef.current == null) throw new Error("useFileDiffInstance: A FileDiff instance should exist when unmounting");
+			// React may detach refs repeatedly when nested Activity trees disappear.
+			if (instanceRef.current == null) return;
 			instanceRef.current.cleanUp();
 			instanceRef.current = null;
 		}
